doc-content/jbpm-docs/src/main/asciidoc/BPMN2/con_custom-tasks-overview.adoc

```diff
@@ -0,0 +1,216 @@
+
+.Milestone
+A milestone represents a single point of achievement within a process instance. You can use milestones to flag certain events to trigger other tasks or track the progress of the process.
+
+image::BPMN2/bpmn-milestone.png[]
+
+Milestones are useful for Key Performance Indicator (KPI) tracking or for identifying the tasks that are still to be completed. Milestones can occur at the end of a stage in a process or they can be the result of achieving other milestones.
+
+Milestones can reach the following states during process execution:
+
+* `Active`: A milestone condition has been defined for the milestone node but it has not been met.
+* `Completed`: A milestone condition has been met (if applicable), the milestone has been achieved, and the process can proceed to the next task or can end.
+
+You can click *Assignments* in the *Properties* panel to open the *Milestone Data I/O* window. In the *Milestone Data I/O* window, you can configure the data input and output as required. For example, to execute a milestone, enter the following data inputs in *Data Inputs and Assignments* fields:
+
+* *Condition*: Condition for the milestone to meet. For example, you can enter a Java expression (string data type) that uses a process variable.
+
+.DecisionTask
+A decision task is used to execute a DMN diagram and invoke a decision engine service from a process. By default, a decision task maps to the DMN decision.
+
+image::BPMN2/bpmn-decision-task-custom.png[]
+
+You can use decision tasks to make an operational decision in a process. Decision tasks are useful for identifying key decisions in a process that need to be made.
+
+You can click *Assignments* in the *Properties* panel to open the *Decision Task Data I/O* window. In the *Decision Task Data I/O* window, you can configure the data input and output as required. For example, to execute a decision task, enter the following data inputs in *Data Inputs and Assignments* fields:
+
+* *Decision*: Decision for a process to make.
+* *Language*: Language of the DMN model, such as `FEEL` or `S-FEEL`.
+* *Model*: Name of the DMN model.
+* *Namespace*: Namespace of the DMN model.
+
+.BusinessRuleTask
+A business rule task is used to evaluate a DRL rule and invoke a decision engine service from a process. By default, a business rule task maps to the DRL rules.
+
+image::BPMN2/bpmn-business-rule-custom-task.png[]
+
+You can use business rule tasks to evaluate key business rules in a business process. You can click *Assignments* in the *Properties* panel to open the *Business Rule Task Data I/O* window. In the *Business Rule Task Data I/O* window, you can configure the data input and output as required. For example, to execute a business rule task, enter the following data inputs in *Data Inputs and Assignments* fields:
+
+* *KieSessionName*: Name of the KIE session.
+* *KieSessionType*: Type of the KIE session.
+* *Language*: Language of the business rule.
```

Comment:
As with DMN, based on the code here:
https://github.com/kiegroup/jbpm/blob/402894e7b9e5cd1ba2efcf31ac62224ec1524472/jbpm-workitems/jbpm-workitems-bpmn2/src/main/java/org/jbpm/process/workitem/bpmn2/AbstractRuleTaskHandler.java#L130-L161
The only possibility is to use DRL if we want to execute a DRL rule.
Also, this value is a default one for the Business Rule Task, so `Language` parameter doesn't have to be specified.

Reply:
Done, changed to _Language of the business rule, defaults to DRL._

Reply:
Same here, it is the language/implementation of the Business Rule Task.

Reply:
Done, changed to _Language of the business rule task, defaults to DRL._